src/containers/layouts/ecommerce/productDetails/byu.modal.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {
+  View,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ListRenderItemInfo,
+  Dimensions,
+} from 'react-native';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import {
+  List,
+  Button,
+} from '@kitten/ui';
+import { Text } from '@src/components/common';
+import { Product as ProductModel } from '@src/core/model';
+import { BuyActionModel } from './productDetails.container';
+
+const { width, height } = Dimensions.get('window');
+
+interface ComponentProps {
+  product: ProductModel;
+  shoppingItemsCount: number;
+  actionItems: BuyActionModel[];
+  onCancel: () => void;
+}
+
+export type ByuModalProps = ThemedComponentProps & ComponentProps;
+
+class ByuModalComponent extends React.Component<ByuModalProps> {
+
+  private onCancel = (): void => {
+    this.props.onCancel();
+  };
+
+  private renderAction = (info: ListRenderItemInfo<BuyActionModel>): React.ReactElement<TouchableOpacityProps> => {
+    const { themedStyle } = this.props;
+
+    return (
+      <TouchableOpacity
+        style={themedStyle.actionContainer}
+        activeOpacity={0.95}
+        onPress={info.item.action}>
+        <Text style={themedStyle.actionLabel}>{info.item.title}</Text>
+      </TouchableOpacity>
+    );
+  };
+
+  public render(): React.ReactNode {
+    const {
+      themedStyle,
+      shoppingItemsCount,
+      product,
+      actionItems,
+    } = this.props;
+
+    return (
+      [
```

Comment:
why is this an array?

Reply:
Just backdrop case. There is not yet implemented customization backdrop in the RNUK